scripts/backfill_api_calls_auth.rb

```
@@ -0,0 +1,15 @@
+# usage: bundle exec rails runner scripts/backfill_api_calls_auth.rb
+
+ApiCall.where(authentication_type: nil).find_each do |api_call|
+  headers = api_call.raw_http.delete("headers")
+
+  if headers["HTTP_AUTHORIZATION"]
+    api_call.authentication_type = "OAuth"
+  elsif headers["HTTP_ACCESS_TOKEN"] && headers["HTTP_UID"]
+    api_call.authentication_type = "DeviseTokenAuth"
+  elsif headers["X-Agent-Auth-Signature"]
+    api_call.authentication_type = "SharedSecret"
+  end
+
+  api_call.save
```

je pense que ça mérite un `.save!` ici pour ne pas avoir de silent fail qui pourraient laisser les headers en clair, non ?

---

j'y ai pensé, et je préfère ne pas avoir d'échec au lancement du script, mais pouvoir ensuite faire un `ApiCall.where(authentication_type: nil)` pour trouver tous les records qui auront des éventuels problèmes